### DISCUSSION TITLE
BashOperator can't find git command

### DISCUSSION THREAD

**Discussion by syed-gilani** [apache/airflow#28385] [Q&A]
I have added a task to a DAG that clones a git repository using BashOperator. Task fails as bash can't find git command. I am running airflow on a kubernetes cluster in Rancher and I have installed airflow using bitnami helm chart.

**Comment by potiuk** [comment]
Check if git is installed (with bitnami) and if it is in the path. We do not manage bitnami chart/image so it's hard to say what is installed there. We have official chart/images in Airflow community (see the docs) and we could help more there and check, but since you rely on bitnami's published artifacts, I guess reaching out to them is better idea.

**Reply by syed-gilani** [reply]
Git is not installed in the airflow worker. I have opened a ticket with bitnami as I think that the Docker image they use for worker doesn't include git. In the mean while I tried installing airflow using the official helm airflow chart https://airflow.apache.org/docs/helm-chart/stable/index.html
on the kubernetes cluster in Rancer. I tried installing it will all the default configurations but it failed with a timeout error waiting for condition. The timeout interval was 600 seconds. I tried it a few times with the same timeout but it failed with the same error on all attempts.

**Reply by potiuk** [reply]
That's likely a Rancer problem that it disabled Helm Hooks:

You can see this for a likely solution: https://airflow.apache.org/docs/helm-chart/stable/index.html#installing-the-chart-with-argo-cd-flux-or-terraform

And if you confirm back I would appreciate submitting additional  comment about Racer's compatibility there - Opening a PR with documentation update explaining that this also applies to Rancer would be most appropriate (and good way to give back to the community for a free software - same as vast majority of our 2300 contributors who are users and found things that they could improve our docs. 

It's super easy - just click "Suggest a change on this page" in the docs and it will open a PR where you will be able to make a PR to update the docs.

**Reply by syed-gilani** [reply]
yes, That fixed the issue. I created the PR https://github.com/apache/airflow/pull/28416

I am now running the same task on airflow instance created from apache airflow helm chart and I get the same error that says git command not found. Please see the image below
<img width="1787" alt="image" src="https://user-images.githubusercontent.com/59292422/208175282-bef418a9-22a1-4443-8677-b3cf2722c848.png">

Also, I get the same git command not found error when I run it in worker shell of the airflow instance. See the image below

<img width="1767" alt="image" src="https://user-images.githubusercontent.com/59292422/208176303-260026e8-0105-4a20-a295-09a375dfc330.png">

**Reply by potiuk** [reply]
Because I guess likely airflow image does not have Git added neither.  Did you check if git is installed in the image? 
Airflow image is minimal to suppor tt . You need to extend the image as described in https://airflow.apache.org/docs/docker-stack/build.html#quick-start-scenarios-of-image-extending and you can add git there, push your own image to your registry and use it from there.

This is how docker images work - they have some "fixed" set of stuff installed and if they are not, you need to build your own image.

**Reply by syed-gilani** [reply]
But how do I reference that image from the airflow helm chart? I am using the airflow helm chart to install airflow on kubernetes cluster. Is there a way to update the helm chart to pull my updated airflow image?

**Reply by potiuk** [reply]
You will find examples in the docs (and reading and understanding configuration parameters of helm chart is a good idea) https://airflow.apache.org/docs/helm-chart/stable/quick-start.html#extending-airflow-image

**Reply by potiuk** [reply]
And again. If you think it was difficult to find - simply add a PR in the place you looked for it and did not found. It's extremely difficult to know how people who do not know will search for such information and you are the BEST person to know where you looked for it and where you did not find it and how to add it so that people like you will be able to find it.

Looking forward to the next PR of yours making it easier to find.

**Reply by syed-gilani** [reply]
Here is the PR to add git as a runtime dependency https://github.com/apache/airflow/pull/28427
